### PR TITLE
Nova Scotia Disability amount

### LINKED ISSUE #442: Nova Scotia Disability amount
Reference :
https://hr.acadiau.ca/files/sites/hr/Payroll/Pensions%20&%20Benefits/NS_TD1_2022.pdf#page=1

### PR DESCRIPTION
**Pull request by okeyiii**
Fixes #442

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Add disability amount policy for Nova Scotia.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Nova Scotia disability amount.
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+description: Nova Scotia disability amount.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml:1]:
```suggestion
description: Nova Scotia provides the following disability amount.
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+values: 
+  2022-01-01: 7_341
+metadata:
+  unit: currency-CAD
+  label: Nova Scotia disability amount.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml:7]:
```suggestion
  label: Nova Scotia disability amount
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+description: Nova Scotia disability amount.
+
+values: 
+  2022-01-01: 7_341
+metadata:
+  unit: currency-CAD
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml:6]:
```suggestion
  unit: currency-CAD
  period: year
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  label: Nova Scotia disability amount.
+  reference: 
+  - title: Nova Scotia Disability Amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml:9]: add legal code (Nova Scotia income tax act)

Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml:9]:
```suggestion
  - title: 2022 Nova Scotia Personal Tax Credits Return - TD1NS
```

File: policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_disability_amount.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    label = "Nova Scotia Disability Amount"
+    definition_period = YEAR
+    defined_for = ProvinceCode.NS
+    reference = "https://hr.acadiau.ca/files/sites/hr/Payroll/Pensions%20&%20Benefits/NS_TD1_2022.pdf#page=1"
+
+    def formula(person, period, parameters):
+        p = parameters(
+            period
+        ).gov.provinces.ns.tax.income.credits.disability_amount
+        isdisabled = person("person_disability_situation", period)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_disability_amount.py:16]:
```suggestion
        is_disabled = person("is_disabled", period)
```

File: policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_disability_amount.py [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+        disability_amount = p.amount
+
+        return where(isdisabled, disability_amount, 0)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/ns/tax/income/credits/ns_disability_amount.py:19]:
```suggestion
        return where(is_disabled, disability_amount, 0)
```

File: policyengine_canada/tests/gov/provinces/ns/tax/income/credits/ns_disability_amount.yaml [hunk 8]
```diff
@@ -0,0 +1,15 @@
+- name: Person is disabled
+  period: 2023
+  input:
+    province_code: NS
+    person_disability_situation: true
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/ns/tax/income/credits/ns_disability_amount.yaml:5]:
```suggestion
    is_disabled: true
```

File: policyengine_canada/tests/gov/provinces/ns/tax/income/credits/ns_disability_amount.yaml [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  output:
+    ns_disability_amount: 7_341
+
+- name: Person is not disabled
+  period: 2023
+  input:
+    province_code: NS
+    person_disability_situation: false
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/ns/tax/income/credits/ns_disability_amount.yaml:13]:
```suggestion
    is_disabled: false
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+description: Nova Scotia provides the following disability amount.
+
+values: 
+  2022-01-01: 7_341
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Nova Scotia disability amount
+  reference: 
+  - title: 2022 Nova Scotia Personal Tax Credits Return - TD1NS
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml:10]: need law reference

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml [hunk 11]
```diff
@@ -0,0 +1,17 @@
+description: Nova Scotia provides the following disability amount.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml:1]:
```suggestion
description: Nova Scotia provides the following base disability amount.
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml [hunk 12]
```diff
@@ -0,0 +1,17 @@
...
+
+values: 
+  2022-01-01: 7_341
+metadata:
+  unit: currency-CAD
+  period: year
+  label: Nova Scotia disability amount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml:8]:
```suggestion
  label: Nova Scotia disability amount base
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+  reference: 
+  - title: 2022 Nova Scotia Personal Tax Credits Return - TD1NS
+    href: https://hr.acadiau.ca/files/sites/hr/Payroll/Pensions%20&%20Benefits/NS_TD1_2022.pdf#page=1
+  - title: T2201 Disability Tax Credit Certificate and relevant information
+    href: https://www.canada.ca/en/revenue-agency/services/forms-publications/forms/t2201.html
+  - title: Income Tax Act 
+    # There is no specific part for disability credits
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml:15]:
```suggestion
    # The income tax act does not specify the disability amount
```

File: policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+  period: year
+  label: Nova Scotia disability amount
+  reference: 
+  - title: 2022 Nova Scotia Personal Tax Credits Return - TD1NS
+    href: https://hr.acadiau.ca/files/sites/hr/Payroll/Pensions%20&%20Benefits/NS_TD1_2022.pdf#page=1
+  - title: T2201 Disability Tax Credit Certificate and relevant information
+    href: https://www.canada.ca/en/revenue-agency/services/forms-publications/forms/t2201.html
+  - title: Income Tax Act 
+    # There is no specific part for disability credits 
+    # But NS government do have T2201 form for supporting it
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/ns/tax/income/credits/disability_amount/amount.yaml:16]:
```suggestion
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/tuition_amount/eligible_age.yaml [hunk 15]
```diff
@@ -0,0 +1,10 @@
+description: British Columbia qualifies filers for the tuition credit of this age or above.
```
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/tuition_amount/eligible_age.yaml:1]: This source is for a federal program though. Does BC inherit the age limitation from Canada rules? Please add a legal code reference to check.
```suggestion
description: British Columbia limits the tuition credit to filers this age or older.
```

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/tuition_amount/eligible_age.yaml
Comment (MaxGhenis) [inline policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/tuition_amount/eligible_age.yaml:1]: why is this in this pr?

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/tuition_amount/eligible_age.yaml (hunk 15 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/bc/tax/income/credits/tuition_amount/eligible_age.yaml:1]: Should not be in this pr, deleted

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**